src/main/java/com/example/demo/chat/manger/DeploymentModelManager.java

```diff
@@ -0,0 +1,27 @@
+package com.example.demo.chat.manger;
+
+import com.example.demo.chat.client.DeploymentNameClient;
+import com.example.demo.chat.config.OpenAiProperties;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Random;
+
+@Component
+@RequiredArgsConstructor
+public class DeploymentModelManager {
+
+    private final OpenAiProperties openAiProperties;
+
+    private final DeploymentNameClient deploymentNameClient;
+
+    public OpenAiProperties updateModel() {
+        List<String> deploymentNames = deploymentNameClient.fetchDeploymentNames();
+        Random random = new Random();
```

Comment:
good for tests to randomly pick the model, but it would probably be better to let the client choose the model :)